presidio-analyzer/presidio_analyzer/predefined_recognizers/country_specific/thai/th_tnin_recognizer.py

```diff
@@ -0,0 +1,165 @@
+from typing import List, Optional, Tuple, Union
+
+from presidio_analyzer import EntityRecognizer, Pattern, PatternRecognizer
+
+
+class ThTninRecognizer(PatternRecognizer):
+    """
+    Recognize Thai National ID Number (TNIN).
+
+    The Thai National ID Number (TNIN) is a unique 13-digit number
+    issued to all Thai residents.
+
+    The format is N1N2N3N4N5N6N7N8N9N10N11N12N13 where:
+    - N1-N12 are the main digits
+    - N13 is a check digit calculated using the preceding 12 digits
+
+    Validation rules:
+    - Must be exactly 13 digits
+    - First digit (N1) cannot be 0
+    - Second digit (N2) cannot be 0
+    - Second and third digits (N2N3) cannot be: 28, 29, 59, 68, 69, 78, 79, 87, 88, 89, 97, 98, 99
+    - The 13th digit is a checksum computed modulo 11 from the first 12 digits
+
+    Checksum algorithm:
+    - Label first 12 digits N1…N12 (left to right)
+    - Compute S = 13·N1 + 12·N2 + … + 2·N12
+    - Let x = S mod 11
+    - Then check digit N13 = (11 − x) mod 10
+    - Equivalently: if x ≤ 1 then N13 = 1 − x; otherwise N13 = 11 − x
+
+    Reference: https://th.wikipedia.org/wiki/เลขประจำตัวประชาชนไทย
+               https://th.wikipedia.org/wiki/ISO_3166-2:TH 
+
+    :param patterns: List of patterns to be used by this recognizer
+    :param context: List of context words to increase confidence in detection
+    :param supported_language: Language this recognizer supports
+    :param supported_entity: The entity this recognizer can detect
+    :param replacement_pairs: List of tuples with potential replacement values
+    for different strings to be used during pattern matching.
+    """
+
+    PATTERNS = [
+        Pattern(
+            "TNIN (Medium)",
+            r"\b\d{13}\b",
+            0.5,
```

Comment:
This would make every 13 digit number a TNIN with score 0.5. Is there a way to embed the different constraints (first/second digits cannot be 0, Second/third cannot be 28, 29, ...) into the regex? This would make the regex much stronger. The alternative is to have the score set as very low, but it would still be predicted if a user uses a threshold of 0.